Metric properties of real negation, absolute value, addition and maximum (#1398)
UniMath/agda-unimath: src/metric-spaces/dependent-products-metric-spaces.lagda.md

  {l l1 l2 : Level} (A : UU l) (P : A → Metric-Space l1 l2) (a : A)
  where

>   is-short-eval-Π-Metric-Space :
>     is-short-function-Metric-Space
>       ( Π-Metric-Space A P)
>       ( P a)
>       ( λ f → f a)

Reviewer: The naming `ev` is actually more consistent with the library, as this is the name we have given the evaluation function in `foundation.evaluation-functions`. I.e., you could have written the following here:

```suggestion
  is-short-ev-Π-Metric-Space :
    is-short-function-Metric-Space
      ( Π-Metric-Space A P)
      ( P a)
      ( ev a)
```

Reviewer: Is there a reason `ev` doesn't work for you here?

Author: > Is there a reason `ev` doesn't work for you here?

Not at all. I was too zealous and was trying to anticipate a comment like "`ev` is too short and not informative enough, please spell out `eval` 😅 . I'll go back to `ev`

  is-saturated-Π-is-saturated-Metric-Space ε x y H a =
    Π-saturated a ε (x a) (y a) (λ d → H d a)
```

### The partial applications of a Cauchy approximation in a dependent product metric space are Cauchy approximations

```agda
module _
  {l l1 l2 : Level} (A : UU l) (P : A → Metric-Space l1 l2)
  (f : cauchy-approximation-Metric-Space (Π-Metric-Space A P))
  where

>   eval-cauchy-approximation-Π-Metric-Space :

Reviewer: ev

    (x : A) →
    is-limit-cauchy-approximation-Metric-Space
      ( P x)
      ( eval-cauchy-approximation-Π-Metric-Space A P f x)
      ( g x)
  is-pointwise-limit-is-limit-cauchy-approximation-Π-Metric-Space L x ε δ =
    L ε δ x

  is-limit-is-pointwise-limit-cauchy-approximation-Π-Metric-Space :
    ( (x : A) →
      is-limit-cauchy-approximation-Metric-Space
        ( P x)
        ( eval-cauchy-approximation-Π-Metric-Space A P f x)
        ( g x)) →
    is-limit-cauchy-approximation-Metric-Space
      ( Π-Metric-Space A P)
      ( f)
      ( g)
  is-limit-is-pointwise-limit-cauchy-approximation-Π-Metric-Space L ε δ x =
    L x ε δ
```

### A product of complete metric spaces is complete

```agda
module _
  {l l1 l2 : Level} (A : UU l) (P : A → Metric-Space l1 l2)
  (Π-complete : (x : A) → is-complete-Metric-Space (P x))
  where

  limit-cauchy-approximation-Π-is-complete-Metric-Space :
    cauchy-approximation-Metric-Space (Π-Metric-Space A P) →
    type-Π-Metric-Space A P
  limit-cauchy-approximation-Π-is-complete-Metric-Space u x =
    limit-cauchy-approximation-Complete-Metric-Space
      ( P x , Π-complete x)
      ( eval-cauchy-approximation-Π-Metric-Space A P u x)

>   is-limit-limit-cauchy-approximation-Π-is-complete-Metric-Space :

Reviewer: ```suggestion
  is-limit-limit-cauchy-approximation-Π-Metric-Space :
```

    (u : cauchy-approximation-Metric-Space (Π-Metric-Space A P)) →
    is-limit-cauchy-approximation-Metric-Space
      ( Π-Metric-Space A P)
      ( u)
      ( limit-cauchy-approximation-Π-is-complete-Metric-Space u)
  is-limit-limit-cauchy-approximation-Π-is-complete-Metric-Space u ε δ x =
    is-limit-limit-cauchy-approximation-Complete-Metric-Space
      ( P x , Π-complete x)
      ( eval-cauchy-approximation-Π-Metric-Space A P u x)
      ( ε)
      ( δ)

>   is-complete-Π-is-complete-Metric-Space :

Reviewer: ```suggestion
  is-complete-Π-Metric-Space :
```
